azure-pipelines.yml

```diff
@@ -189,7 +189,7 @@ stages:
             pip install -U -r requirements.txt -r requirements-dev.txt -c constraints.txt
             python setup.py sdist
             pip install -U -c constraints.txt dist/qiskit-terra*.tar.gz
-            pip install -U "qiskit-aer" "z3-solver" -c constraints.txt
+            pip install -U "cplex" "qiskit-aer" "z3-solver" -c constraints.txt
```

Comment:
we don't need `docplex` here? why is it needed in setup.py but not here?

Reply:
`docplex` is a free library, so we included it in `requirements-dev.txt` (that's why we don't need `docplex` here)